api/topology/v1/topology_api.proto

```diff
@@ -31,6 +40,54 @@ message GetTopologyResponse {
   repeated QueryResult results = 1;
 }
 
+message SearchTopologyRequest {
+  message FieldSelector {
+    enum Column {
+      UNSPECIFIED = 0;
+      ID = 1;
+    }
+    oneof field {
+      Column column = 1;
+      // if metadata is nested specific the full path like so
+      // meatadata = 'level1.level2.level3.fieldname'
+      string metadata = 2;
+    }
+  }
+
+  // The default sort is by column `id` and descending
+  message Sort {
+    enum Direction {
+      UNSPECIFIED = 0;
+      ASCENDING = 1;
+      DESCENDING = 2;
+    }
+    Direction direction = 1;
+    FieldSelector field = 2 [ (validate.rules).message.required = true ];
+  }
+
+  message Filter {
+    message Search {
+      FieldSelector field = 1 [ (validate.rules).message.required = true ];
+      string text = 2 [ (validate.rules).string.min_len = 1 ];
+    }
+
+    Search search = 1;
+    string type_url = 2;
+    // Any valid metadata on the cache Resource object
+    map<string, string> metadata = 3;
+  }
+
+  Sort sort = 1;
+  string page_token = 2;
```

Comment:
Chatted IRL - only thing to follow up on is clarity around the default behavior of the page token here

Reply:
for reference: https://cloud.google.com/apis/design/design_patterns#list_pagination